chain/default_store.go

```diff
@@ -391,13 +391,18 @@ func (store *DefaultStore) BlockHeight() (uint64, error) {
 	return store.head.Height()
 }
 
-// ActorFromLatestState gets the latest state and retrieves an actor from it.
-func (store *DefaultStore) ActorFromLatestState(ctx context.Context, addr address.Address) (*actor.Actor, error) {
+// LatestState gets the latest state from the state Store.
+func (store *DefaultStore) LatestState(ctx context.Context) (state.Tree, error) {
 	tsas, err := store.GetTipSetAndState(store.GetHead())
 	if err != nil {
 		return nil, err
 	}
-	st, err := state.LoadStateTree(ctx, store.stateStore, tsas.TipSetStateRoot, builtin.Actors)
+	return state.LoadStateTree(ctx, store.stateStore, tsas.TipSetStateRoot, builtin.Actors)
+}
+
+// ActorFromLatestState gets the latest state and retrieves an actor from it.
+func (store *DefaultStore) ActorFromLatestState(ctx context.Context, addr address.Address) (*actor.Actor, error) {
```

Comment:
We shouldn't be putting these on the chain store.  I don't know how actor got here but it doesn't belong here.  Long ago I put this exact `LatestState` on the function and @phritz brought up the good point that this helps make the api more redundant and adds more opportunity to footgun.

With my `LatestState` api back in the day we ended up having a tricky bug because `store.LatestState` and `store.GetTsas(store.Head())` did not match.

If we are trying to clean this up (which is a great idea) my strong opinion is we need to reduce the set of calls on the store (push them up to other layers) not add calls to the store.

Reply:
*on the function => on the chain store interface

Reply:
This was added a while ago by @acruikshank. I'm planning to remove it in this PR, but first I needed to get some changes merged in under it from @jscode017.